src/lib.rs

```diff
@@ -106,16 +107,166 @@ impl Build {
+        let mut watcher: RecommendedWatcher =
+            notify::Watcher::new(tx, std::time::Duration::from_secs(2))
+                .context("could not initialize watcher")?;
+
+        let metadata = cargo_metadata::MetadataCommand::new()
+            .exec()
+            .context("cannot get package's metadata")?;
+
+        self.exclude(metadata.target_directory.as_std_path());
+
+        if self.watch_paths.is_empty() {
+            log::trace!("Watching {}", &metadata.workspace_root);
+            watcher
+                .watch(&metadata.workspace_root, RecursiveMode::Recursive)
+                .context("cannot watch this crate")?;
+        } else {
+            for path in &self.watch_paths {
+                match watcher.watch(&path, RecursiveMode::Recursive) {
+                    Ok(()) => log::trace!("Watching {}", path.display()),
+                    Err(err) => log::error!("cannot watch {}: {}", path.display(), err),
+                }
+            }
+        }
+
+        let mut child = command.spawn().context("cannot spawn command")?;
+
+        loop {
+            use notify::DebouncedEvent::*;
+
+            let message = rx.recv();
+
+            match &message {
+                Ok(Create(path)) | Ok(Write(path)) | Ok(Remove(path)) | Ok(Rename(_, path))
+                    if !self.is_excluded_path(path) && !self.is_hidden_path(path) =>
+                {
+                    #[cfg(unix)]
+                    {
+                        let now = std::time::Instant::now();
+
+                        unsafe {
+                            libc::kill(
+                                child.id().try_into().expect("cannot get process id"),
+                                libc::SIGTERM,
+                            );
+                        }
+
+                        while now.elapsed().as_secs() < 2 {
+                            std::thread::sleep(std::time::Duration::from_millis(200));
+                            if let Ok(Some(_)) = child.try_wait() {
+                                break;
+                            }
+                        }
+                    }
+
+                    match child.try_wait() {
+                        Ok(Some(_)) => {}
+                        _ => {
+                            let _ = child.kill();
+                            let _ = child.wait();
+                        }
+                    }
+
+                    child = command.spawn().context("cannot spawn command")?;
+                }
+                Ok(_) => {}
+                Err(err) => log::error!("watch error: {}", err),
+            };
+        }
+    }
+}
+
+impl Default for Watch {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 #[derive(Debug, StructOpt)]
-pub struct DevServer {
+pub struct Serve {
     #[structopt(long, default_value = "127.0.0.1")]
     ip: IpAddr,
     #[structopt(long, default_value = "8000")]
     port: u16,
+    #[structopt(skip)]
+    watch: Watch,
```

Comment:
I said flattening. Not skipping it xD 

If you flatten, the options of Watch will become available to the user which can be useful